src/core/CameraControls.tsx

```diff
@@ -0,0 +1,48 @@
+import * as THREE from 'three'
+import type { PerspectiveCamera, OrthographicCamera } from 'three'
+
+import * as React from 'react'
+import { forwardRef, useMemo, useEffect } from 'react'
+import { extend, useFrame, useThree, ReactThreeFiber, EventManager } from '@react-three/fiber'
+
+import CameraControlsImpl from 'camera-controls'
+
+CameraControlsImpl.install({ THREE })
+extend({ CameraControlsImpl })
```

Comment:
These break tree-shaking, so I'd try to move them into the component as a render-effect like we do in R3F's canvas: https://github.com/pmndrs/react-three-fiber/blob/b06ff7478f53166e5cf65402a43347f6b32d453c/packages/fiber/src/web/Canvas.tsx#L52-L55

This may be a problem elsewhere in Drei, so apologies if you're just following convention.

Reply:
>  I'd try to move them into the component as a render-effect like we do in R3F's canvas: https://github.com/pmndrs/react-three-fiber/blob/b06ff7478f53166e5cf65402a43347f6b32d453c/packages/fiber/src/web/Canvas.tsx#L52-L55
apart from that (I'm not sure to understand the fix you suggest)

Would defining a `subsetOfTHREE` object like explained here https://github.com/yomotsu/camera-controls#important

```ts
const subsetOfTHREE = {
	Vector2   : Vector2,
	Vector3   : Vector3,
	Vector4   : Vector4,
	Quaternion: Quaternion,
	Matrix4   : Matrix4,
	Spherical : Spherical,
	Box3      : Box3,
	Sphere    : Sphere,
	Raycaster : Raycaster,
	MathUtils : {
		DEG2RAD: MathUtils.DEG2RAD,
		clamp: MathUtils.clamp,
	},
};

CameraControls.install( { THREE: subsetOfTHREE } );
```

could help?

I can easily do this, if you confirm ;)

Reply:
It wouldn't solve the tree-shaking issue -- that has to be moved out of global/module scope, but it'll pull in a lot less of three.js when the component is imported. If the author doesn't anticipate this subset changing between versions, it would be safe to do, yeah.

Reply:
better with your snippet now, I will make this change

Reply:
ok done https://github.com/pmndrs/drei/pull/1237/files#diff-a0d895145d4d77f878604f353413ea046de8817965da0f2dad997985d06fe842R22-R25